docs/extensions/sdk-composables.md

```diff
@@ -0,0 +1,132 @@
+---
+description: Learn about the extension SDK composables and how to utilize them when developing custom extensions.
+contributors: Esther Agbaje
+---
+
+# Extension SDK Composables
+
+The Extension Composables serve as a powerful toolkit to help developers create extensions in Directus.
+
+Rather than needing to rewrite logic from scratch, extension developers can leverage primitives like `useApi()` or
+`useStores()`, to abstract away complexity when building extensions.
+
+## useApi()
+
+The `useApi` composable allows extensions to make requests to endpoints within Directus. It acts as a wrapper around the
+axios library.
+
+Here’s a sample code of how to implement `useApi`
+
+```ts
+<script setup>
+import { useApi } from '@directus/extensions-sdk';
+
+const api = useApi();
+
+async function fetchData() {
+      const response = await api.get(ENDPOINT-URL);
+      data.value = response.data;
+    }
+fetchData();
+</script>
+```
+
+## useStores()
+
+`useStores` serves as the main entry point for accessing specific functionality when building extensions. Within
+`useStores` are sub-stores like the `usePermissionsStore`, `useCollectionsStore` and `useFieldsStore` among others.
+
+```ts
+<script setup>
+import { useStores } from '@directus/extensions-sdk';
+
+const { useFieldsStore, usePermissionsStore, useCollectionStore } = useStores();
+const fieldsStore = useFieldsStore();
+const permissionsStore = usePermissionsStore();
+const collectionStore = useCollectionStore();
+</script>
+```
+
+### useFieldsStore
+
+The `useFieldsStore` gives developers access to customizable field properties and functionality. It provides methods
+like `fields`, `updateField` , and `getPrimaryKeyFieldForCollection`.
+
+For a deep dive on how to use the `useFieldsStore` composable,
+[see the implementation](https://github.com/directus/directus/blob/main/app/src/stores/fields.ts#L69) in our codebase.
```

Comment:
This is missing the 'why'. WHY does this exist? WHEN would I use it? Same for all items.